### PR TITLE
Project 15: Todo list

### PR DESCRIPTION
**Pull request by Thessan**
**Link to my deployed project:** [https://tessans-todo.netlify.app](https://tessans-todo.netlify.app)

**My reflections on how this week's project turned out:** Was a bit tricky to get started and I had to spend a lot of time going back to reading up on things. There are things I want to look up more and add in the future, but I'm happy with what I've accomplished so far. 

Have not yet designed it for desktop, the site looks best in mobile version (iPhoneX / Chrome inspect)

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,57 @@
...
-    </div>
+    /* all components called within the Provider below will have access
+    to the data stored in savedData */
+    <Provider store={store}> 
+        <Wrapper>
+          <Header />
+            <AddNewTodo />
+              <TodoList />  
+                <Counter />
+        </Wrapper>
```
Comment (Corneliaryas) [inline code/src/App.js]: Maybe set the indentation so that Header, AddNewTodo, TodoList and Counter is on the same level?

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,57 @@
...
+};
+
+/*  the 'original' setup for the reducer
+( const store = configureStore({ reducer }); )
+is being changed to the line of code below */
+const store = createStore(reducer, persistedState, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
+
+store.subscribe(() => {
+  localStorage.setItem("savedData", JSON.stringify(store.getState()));
+});
```
Comment (Corneliaryas) [inline code/src/App.js:34]: So cool that you are using local storage, makes the app so much more useful! 👍😊

File: code/src/components/AddNewTodo.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                required
+                onChange={event => setNewTodo(event.target.value)}
+            />
+
+            <AddTodoButton type="submit"
+                onClick={() => {
+                    onSubmit();
+                    setNewTodo(''); /* clears input field after adding a new task */
+                }}
+                disabled={ newTodo.length < 5 }>
```
Comment (Corneliaryas) [inline code/src/components/AddNewTodo.js]: Nice with the disabled button! 👍

File: code/src/components/AddNewTodo.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+font-weight: bold;
+`
+
+const Input = styled.input`
+background-color: #F4F4F4;
+font-family: 'Josefin Sans', sans-serif;
+width: 150px;
+border: none;
+border-bottom: solid 2px #baa8a8;
+`
```
Comment (Corneliaryas) [inline code/src/components/AddNewTodo.js:64]: Nice use of styled components and styled form input!

File: code/src/components/Counter.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+display: flex;
+justify-content: center;
+background-color: #F4F4F4;
+padding-top: 20px;
+padding-bottom: 20px;
+`
+
+const CounterText = styled.text`
+font-size: 18px;
+`
```
Comment (Corneliaryas) [inline code/src/components/Counter.js:28]: Clean code with these separate components! 👍

File: code/src/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    const mm = date.getMonth();
+    const month = months[mm];
+    const dd = date.getDate();
+    const weekday = date.toLocaleDateString('en-US', {weekday: 'long'});
+
+    return (
+        <HeaderContainer>
+            <Image src={headerImage} />
+                <HeaderText>MY TO-DO'S</HeaderText>
+                    <DateText>{weekday}, {month} {dd}</DateText>
```
Comment (Corneliaryas) [inline code/src/components/Header.js:20]: So nice header with today's date and the background image!

File: code/src/components/TodoList.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            {items.map(item => (
+        <TodoContainer key={item.id}>
+          <List>
+      <label>
+          <input type="checkbox"
+          checked={item.complete}
+          onChange={() => onChecked(item.id)}
+          />
+      </label>
+  </List>
```
Comment (Corneliaryas) [inline code/src/components/TodoList.js]: A bit of indentation that needs fixing to make the code just a bit more readable 😊

File: code/src/reducers/todos.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    items: []
+}
+
+export const todos = createSlice({
+    name: 'todos',
+    initialState,
+    reducers: {
+        removeItem:(store, action) => { 
+            const itemId = action.payload;
+            const filteredList = store.items.filter(item => item.id !== itemId)
```
Comment (Corneliaryas) [inline code/src/reducers/todos.js:14]: Clever way of deleting the item!

File: code/src/reducers/todos.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+            store.items = filteredList;
+        },
+
+        addNewTodo: (store, action) => {
+            const newItem = {
+                id: uuidv4(), /* adds a uniqe id for each ToDo */
```
Comment (Corneliaryas) [inline code/src/reducers/todos.js]: So smart, had no idea about this!

**Review by Corneliaryas — APPROVED** [review]
Great job Therése! Your site looks really great, love the design (!) and works just as expected! A big plus for the local storage!! 😊🎉

Just found some minor things to feedback on, mostly indentation 😊

(Sorry for the late review!!)

File: code/src/components/AddNewTodo.js (hunk 3 above)
Comment (HIPPIEKICK) [reply code/src/components/AddNewTodo.js]: But what if I want to "code"? ;)

File: code/src/components/AddNewTodo.js (hunk 4 above)
Comment (HIPPIEKICK) [reply code/src/components/AddNewTodo.js:64]: Just remember indenting the styled components as well

File: code/src/components/Counter.js (hunk 5 above)
Comment (HIPPIEKICK) [reply code/src/components/Counter.js:28]: Is there an HTML element called text I didn't know about? ;)

File: code/src/components/AddNewTodo.js [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+                value={newTodo}
+                placeholder="Add new todo..."
+                required
+                onChange={event => setNewTodo(event.target.value)}
+            />
+
+            <AddTodoButton type="submit"
+                onClick={() => {
+                    onSubmit();
+                    setNewTodo(''); /* clears input field after adding a new task */
```
Comment (HIPPIEKICK) [inline code/src/components/AddNewTodo.js]: Semi colon? 🤔

File: code/src/components/Header.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+const HeaderContainer = styled.section `
+display: flex;
+justify-content: center;
+align-items: center;
+position: relative;
+width: 100%;
+height: auto;
+`
```
Comment (HIPPIEKICK) [inline code/src/components/Header.js]: Quick fix for a (little bit) better look on desktop:
a `max-height` and `object-fit: cover;` ⭐

**Comment by HIPPIEKICK** [comment]
Go through the indentation in all your files as it makes your otherwise mostly clean written project look a little messy :)

File: code/src/components/AddNewTodo.js (hunk 3 above)
Comment (Thessan) [reply code/src/components/AddNewTodo.js]: changed it to 4 letters 😄

File: code/src/components/AddNewTodo.js (hunk 4 above)
Comment (Thessan) [reply code/src/components/AddNewTodo.js:64]: fixed! 👍

File: code/src/components/Counter.js (hunk 5 above)
Comment (Thessan) [reply code/src/components/Counter.js:28]: apparently? :sweat_smile: hehe, changed it to a **styled.p**

File: code/src/components/AddNewTodo.js (hunk 10 above)
Comment (Thessan) [reply code/src/components/AddNewTodo.js]: deleted!

**Pull request closed without merging**